test/lib/actions/RulesAction.test.ts

```diff
@@ -1,7 +1,90 @@
-import {RulesAction} from '../../../src/lib/actions/RulesAction';
+import {RulesAction, RulesDependencies} from '../../../src/lib/actions/RulesAction';
+import {StubDefaultConfigLoader} from '../../stubs/StubConfigLoader';
+import * as StubEngineLoaders from '../../stubs/StubEngineLoaders';
+import {StubRuleViewer} from '../../stubs/StubRuleViewer';
 
 describe('RulesAction tests', () => {
-	it('Invoke #execute() to enforce coverage', () => {
-		new RulesAction().execute({});
+
+	it('Submitting the all-selector returns all rules', () => {
+		const action = new RulesAction();
+		const viewer = new StubRuleViewer();
+		const dependencies: RulesDependencies = {
+			configLoader: new StubDefaultConfigLoader(),
+			engineLoader: new StubEngineLoaders.StubEngineLoader_withFunctionalStubEngine(),
+			viewer
+		};
+		const input = {
+			'rule-selector': ['all']
+		};
+
+		action.execute(dependencies, input);
+
+		viewer.expectViewedRules([
+			'stub1RuleA',
+			'stub1RuleB',
+			'stub1RuleC',
+			'stub1RuleD',
+			'stub1RuleE',
+			'stub2RuleA',
+			'stub2RuleB',
+			'stub2RuleC'
+		]);
+	});
+
+	it('Submitting a filtering selector returns only matching rules', () => {
+		const action = new RulesAction();
+		const viewer = new StubRuleViewer();
+		const dependencies: RulesDependencies = {
+			configLoader: new StubDefaultConfigLoader(),
+			engineLoader: new StubEngineLoaders.StubEngineLoader_withFunctionalStubEngine(),
+			viewer
+		};
+		const input = {
+			'rule-selector': ['CodeStyle']
+		};
+
+		action.execute(dependencies, input);
+
+		viewer.expectViewedRules([
+			'stub1RuleA',
+			'stub1RuleD'
+		]);
+	});
+
+	/**
+	 * This behavior was not formally defined, and isn't actually possible at the moment due to
+	 * hard-coded engines. But in the future we may want to have dynamic engine loading, and this
+	 * test will help us do that.
+	 */
+	it('When no engines are registered, empty results are displayed', () => {
```

Comment:
Agreed - always nice to have an edge case test... even if we don't use it - if something goes wrong and empty somehow gets in there we want to make sure things don't blow up - and behave nicely.